### PR TITLE
Power fixes

### PR DESCRIPTION
**Pull request by ChuckM**
This PR updates the pwr.h file and the rcc.c file to deal with a multibit VOS field.

### CODE REVIEW

File: include/libopencm3/stm32/common/flash_common_f234.h [hunk 1]
```diff
@@ -50,7 +50,8 @@
 
 /* --- FLASH_ACR values ---------------------------------------------------- */
 
-#define FLASH_ACR_LATENCY_MASK		0x07
+#define FLASH_ACR_LATENCY_MASK		0x0f
+#define FLASH_ACR_LATENCY(w)		((w) & FLASH_ACR_LATENCY_MASK)
 #define FLASH_ACR_LATENCY_0WS		0x00
 #define FLASH_ACR_LATENCY_1WS		0x01
 #define FLASH_ACR_LATENCY_2WS		0x02
```
Comment (karlp) [inline include/libopencm3/stm32/common/flash_common_f234.h]: I've been inclinded to just drop all of these.  latency is just a natural number.  as long as we have the mask we should be good.  doing it like this is probably good interim step, but in all reality, without versioned tags, there's no way of doing a timed deprecate,remove phase

File: include/libopencm3/stm32/f4/pwr.h [hunk 2]
```diff
@@ -72,14 +73,13 @@ LGPL License Terms @ref lgpl_license
 
 /* --- Function prototypes ------------------------------------------------- */
 
-enum pwr_vos_scale {
-	PWR_SCALE1,
-	PWR_SCALE2,
-};
+#define	PWR_SCALE1	0x3
+#define PWR_SCALE2	0x2
+#define PWR_SCALE3	0x1
```
Comment (karlp) [inline include/libopencm3/stm32/f4/pwr.h]: hrm? what's worng with the enum?  we use an enum on all the other platforms with multi bits, eg f7, l1, l0, (stopped looking)

File: lib/stm32/f4/pwr.c [hunk 3]
```diff
@@ -36,11 +36,9 @@
...
+void pwr_set_vos_scale(int8_t scale)
 {
-	if (scale == PWR_SCALE1) {
-		PWR_CR |= PWR_CR_VOS;
-	} else if (scale == PWR_SCALE2) {
-		PWR_CR &= PWR_CR_VOS;
-	}
+	uint32_t reg = PWR_CR & ~(PWR_CR_VOS_MASK << PWR_CR_VOS_SHIFT);
+	reg |= (scale & PWR_CR_VOS_MASK) << PWR_CR_VOS_SHIFT;
+	PWR_CR = reg;
```
Comment (karlp) [inline lib/stm32/f4/pwr.c]: it does make less code doesn't it...  perhaps the enums with defined values?

File: lib/stm32/f4/rcc.c [hunk 4]
```diff
@@ -59,9 +59,9 @@ const struct rcc_clock_scale rcc_hse_8mhz_3v3[RCC_CLOCK_3V3_END] = {
 		.hpre = RCC_CFGR_HPRE_DIV_NONE,
 		.ppre1 = RCC_CFGR_PPRE_DIV_4,
 		.ppre2 = RCC_CFGR_PPRE_DIV_2,
-		.power_save = 1,
+		.power_save = PWR_SCALE1,
 		.flash_config = FLASH_ACR_DCEN | FLASH_ACR_ICEN |
-				FLASH_ACR_LATENCY_3WS,
+				FLASH_ACR_LATENCY_1WS,
```
Comment (karlp) [inline lib/stm32/f4/rcc.c]: this looks like a standalone bugfix?

File: lib/stm32/f4/rcc.c [hunk 5]
```diff
@@ -75,6 +75,7 @@ const struct rcc_clock_scale rcc_hse_8mhz_3v3[RCC_CLOCK_3V3_END] = {
 		.hpre = RCC_CFGR_HPRE_DIV_NONE,
 		.ppre1 = RCC_CFGR_PPRE_DIV_2,
 		.ppre2 = RCC_CFGR_PPRE_DIV_NONE,
+		.power_save = PWR_SCALE1,
```
Comment (karlp) [inline lib/stm32/f4/rcc.c]: I think we should stop calling this power_save while we're here, it's called ```voltage_scale``` on l1/l0, and ```vos_scale``` everywhere else.  (We should even move l1/l0 to vos_scale as well, but that's out of scope of this)

File: lib/stm32/f4/rcc.c [hunk 6]
```diff
@@ -254,9 +260,9 @@ const struct rcc_clock_scale rcc_hse_25mhz_3v3[RCC_CLOCK_3V3_END] = {
 		.hpre = RCC_CFGR_HPRE_DIV_NONE,
 		.ppre1 = RCC_CFGR_PPRE_DIV_4,
 		.ppre2 = RCC_CFGR_PPRE_DIV_2,
-		.power_save = 1,
+		.power_save = PWR_SCALE1,
 		.flash_config = FLASH_ACR_DCEN | FLASH_ACR_ICEN |
-				FLASH_ACR_LATENCY_3WS,
+				FLASH_ACR_LATENCY_1WS,
```
Comment (karlp) [inline lib/stm32/f4/rcc.c]: another pre-existing bug?

File: lib/stm32/f4/rcc.c (hunk 4 above)
Comment (ChuckM) [reply lib/stm32/f4/rcc.c]: copypasta code for when someone created the 48Mhz struct they didn't update wait states.

File: lib/stm32/f4/pwr.c (hunk 3 above)
Comment (ChuckM) [reply lib/stm32/f4/pwr.c]: either works although when they change it to 3 bits the enum needs to be reworked. The defined values allows code transparency while the changes happen at the loss of compiler ability to detect mis-matched arguments. This function isn't called in a lot of code (certainly on the F4 side)

**Comment by karlp** [comment]
fixed all via https://github.com/libopencm3/libopencm3/pull/949/commits/2649c5e493028b54b67baa6c202ac17a4c9b8951 aiui

**Pull request closed without merging**